srcCpp/Infrastructure_common.cxx

```diff
@@ -57,8 +74,17 @@ void PerftestTimer::setTimeout(
             0,
             0);
   #else
-    signal(SIGALRM, PerftestTimer::timeoutTask);
-    alarm(executionTimeInSeconds);
+    // We have to create a new pointer to the timer so 
+    // it is not removed when this function ends
+    timerThread = PerftestThread_new(
```

Comment:
Is this thread destroyed at some point?
Check `PerftestThread_delete`, this will call OSAPI and finalize the thread properly.

Reply:
Now it is in my last commit ;)